x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/diagnostic/circuit_breakers/rss_grouth_circuit_breaker.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  CircuitBreaker,
+  CircuitBreakerResult,
+} from '../health_diagnostic_circuit_breakers.types';
+import { failure, success } from './utils';
+
+export class RssGrouthCircuitBreaker implements CircuitBreaker {
+  private readonly initialRss: number;
+  private maxRss: number;
+  private maxPercentGrowth: number;
+
+  constructor(
+    private readonly config: {
+      maxRssGrowthPercent: number;
+      validationIntervalMs: number;
+    }
+  ) {
+    if (config.maxRssGrowthPercent < 0 || config.maxRssGrowthPercent > 1) {
+      throw new Error('maxRssGrowthPercent must be between 0 and 1');
```

Comment:
IIUC [rss](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/process.d.ts#L134C63-L134C80) returns
> method returns an integer representing the Resident Set Size (RSS) in bytes.

if so, lets have an example: process uses 1MB memory. It could growth till 10MB, which sill not be an issue on modern hardware, which is 10x. 

My point is: why would limit top growth to 2x (percent = 1)? IMHO the up limit could be up-to the client of the API depending on the env. 

Just FYI.